### PR TITLE
add-resource-params, fix filter, testing on minikube, updated quickstart

### LINKED ISSUE #50: argo workflow trigger with params
**Is your feature request related to a problem? Please describe.**
`Argo-events` was created in order to become the event framework for `Argo`. While we support creating argo `workflow` as part of a trigger, they are encompassed in the more general `ResourceObjects` which makes it difficult to couple valuable features like `Argo` params with the event framework. 

**Describe the solution you'd like**
As discussed in #19 we should have an explicit argo workflow trigger so that we can easily pass arguments/params to the workflow.

**Additional context**
This may help to solve the problem of #41 depending on how the workflows are included in the sensor spec.

### PR DESCRIPTION
**Pull request by magaldima**
closes #50

### CODE REVIEW

File: examples/calendar-sensor.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
         group: argoproj.io
         version: v1alpha1
         kind: Workflow
-        artifactLocation:
+        source:
```
Comment (shrinandj) [inline examples/calendar-sensor.yaml]: +1 for changing this. Argo workflows themselves have a notion of artifacts and overloading that term was making it confusing.

Comment (shrinandj) [reply examples/calendar-sensor.yaml]: In a following change, we should also rename structs from `S3Artifact` to `S3Source` or something like that.

**Comment by magaldima** [comment]
TESTING EVIDENCE (from Minikube):

I created this webhook sensor:
```
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: webhook-with-resource-param
  labels:
    sensors.argoproj.io/controller-instanceid: axis
spec:
  signals:
    - name: webhook
      webhook:
        endpoint: /hello
        port: 7070
        method: POST
  triggers:
    - name: argo-workflow
      resource:
        namespace: default
        group: argoproj.io
        version: v1alpha1
        kind: Workflow
        # The parameters from the workflow are overridden by the webhook's message
        parameters:
          - src:
              signal: webhook
              path: message
              value: hello default
            dest: spec.arguments.parameters.0.value
        source:
          inline: |
              apiVersion: argoproj.io/v1alpha1
              kind: Workflow
              metadata:
                name: arguments-via-webhook-event
              spec:
                entrypoint: whalesay
                arguments:
                  parameters:
                  - name: message
                    # this is the value that should be overridden
                    value: hello world
                templates:
                - name: whalesay
                  inputs:
                    parameters:
                    - name: message
                  container:
                    image: docker/whalesay:latest
                    command: [cowsay]
                    args: ["{{inputs.parameters.message}}"]
```

Then I sent an HTTP POST:
```
{
	"message": "this is my first webhook"
}
```

the logs of the argo workflow:
```
 __________________________
< this is my first webhook >
 --------------------------
    \
     \
      \
                    ##        .
              ## ## ##       ==
           ## ## ## ##      ===
       /""""""""""""""""___/ ===
  ~~~ {~~ ~~~~ ~~~ ~~~~ ~~ ~ /  ===- ~~~
       \______ o          __/
        \    \        __/
          \____\______/
```

File: examples/calendar-sensor.yaml (hunk 1 above)
Comment (magaldima) [reply examples/calendar-sensor.yaml]: agreed

**Comment by magaldima** [comment]
Don't approve yet.. I gotta re-run dep ensure because of the new dependency on `tidwall/sjson`

**Comment by magaldima** [comment]
okay all good now.

**Pull request merged**